### PR TITLE
Add constructor return type annotations

### PR DESCRIPTION
**Pull request by palango**
Also removes the `hypothesis` dependency

### CODE REVIEW

File: raiden/network/rpc/smartcontract_proxy.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ def __init__(
             contract_address: Address,
             call_function: Callable,
             transact_function: Callable,
```
Comment (LefterisJP) [inline raiden/network/rpc/smartcontract_proxy.py]: Why remove the Optiona() from here?

**Comment by LefterisJP** [comment]
Also a general comment @palango: Do we really need to specify that constructors return `None` everywhere? I find it completely redundant and I would not want to have to write the obvious for every constructor I write.

File: raiden/network/rpc/smartcontract_proxy.py (hunk 1 above)
Comment (palango) [reply raiden/network/rpc/smartcontract_proxy.py]: When setting a parameters default to `None`, `mypy` infers that the type is `Optional`.

**Comment by palango** [comment]
I feel that same. It's a problem with mypy at them moment ( see https://github.com/python/mypy/issues/604) 

On the other hand I think it's not a big deal to add it and doesn't hurt readability.

**Comment by LefterisJP** [comment]
I would say not to do this. It's completely unecessary and even guido seems to think so: 

> https://github.com/python/mypy/issues/604#issuecomment-83887491

Let them fix mypy. It's a tool to help us control our typing problems, not to burden our code with unecessary annotations. 

> and doesn't hurt readability.

Unecessary annotations hurt readability

**Comment by palango** [comment]
> Let them fix mypy. It's a tool to help us control our typing problems, not to burden our code with unecessary annotations.

Totally agree, but I'd add the annotations until it's fixed.

**Comment by LefterisJP** [comment]
@palango Since we don't enforce mypy anywhere and it's just used locally to help us I don't want us to add the annotations for return type of constructors.

**Comment by palango** [comment]
I believe it would be worth to reduce the noise of the mypy suggestions, but Ok.

**Pull request closed without merging**